deployments/vanilla/terraform/main.tf

```diff
@@ -0,0 +1,154 @@
+locals {
+  name = "vanilla-kubeflow"
+  cluster_name = coalesce(var.cluster_name, local.name)
+  region       = var.cluster_region
+  eks_version = var.eks_version
+
+  vpc_cidr = "10.0.0.0/16"
+  az_count = local.region == "us-west-1" ? 2 : 3
+  azs      = slice(data.aws_availability_zones.available.names, 0, local.az_count)
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/awslabs/kubeflow-manifests"
+  }
+
+  kf_helm_repo_path = var.kf_helm_repo_path
+}
+
+provider "aws" {
+  region = local.region
+}
+
+provider "kubernetes" {
+  host                   = module.eks_blueprints.eks_cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks_blueprints.eks_cluster_id]
+  }
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.eks_blueprints.eks_cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+
+    exec {
+      api_version = "client.authentication.k8s.io/v1beta1"
+      command     = "aws"
+      # This requires the awscli to be installed locally where Terraform is executed
+      args = ["eks", "get-token", "--cluster-name", module.eks_blueprints.eks_cluster_id]
+    }
+  }
+}
+
+data "aws_availability_zones" "available" {}
+
+#---------------------------------------------------------------
+# EKS Blueprints
+#---------------------------------------------------------------
+module "eks_blueprints" {
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.9.0"
```

Comment:
pull out the tag as a local variable?

Reply:
source values need to be hard coded paths since terraform loads them first and then looks at their contents after (e.g. local vars, resources, etc.)